### PR TITLE
storage: Enable PreVote and TickQuiesced by default

### PR DESCRIPTION
**Pull request by bdarnell**

### CODE REVIEW

File: pkg/storage/replica.go [hunk 1]
```diff
@@ -3767,9 +3769,17 @@ func (r *Replica) tick() (bool, error) {
 		// difficult to completely eliminate them so we want to minimize the
 		// disruption when they do occur.
 		//
-		// For more details, see #9372.
-		// TODO(bdarnell): remove this once we have fully switched to PreVote
-		if !enablePreVote {
+		// Without CheckQuorum, the call to TickQuiesced is less critical,
+		// but still improves our responsiveness to node failures by
+		// priming the replica to start an election immediately upon
```
Comment (nvanbenschoten) [inline pkg/storage/replica.go]: > priming the replica to start an election immediately upon unquiescing

Priming any replica or priming followers? Is this important for a leader?

**Review by nvb — APPROVED** [review]

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/storage/store.go, line 120 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/22316#-L4LdE20NDZMdxzxj8yO:-L4LdE20NDZMdxzxj8yP:b-maaad2) ([raw file](https://github.com/cockroachdb/cockroach/blob/e88b34aea14f3e0f2c63403ecf31b1cf65d13059/pkg/storage/store.go#L120)):*
> ```Go
> 
> var enableTickQuiesced = envutil.EnvOrDefaultBool(
> 	"COCKROACH_ENABLE_TICK_QUIESCED", true)
> ```

When would we want to disable tick quiesced? Per https://github.com/cockroachdb/cockroach/issues/16950#issuecomment-362454601, it seems like additional code will be required to disable tick quiesced, but before that happens disabling quiesced ticking is a bad idea.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22316#-:-L4Ld_k1jZCQ7b2CXQt2:bnfp4nl)*

File: pkg/storage/replica.go (hunk 1 above)
Comment (bdarnell) [reply pkg/storage/replica.go]: With CheckQuorum off, this only affects followers.

**Comment by bdarnell** [comment]
We need additional campaign logic to remove TickQuiesced completely, but until then, there's a chance it's not a win in all cases. As I said in comments in this PR, I think high latency and skewed traffic might benefit from disabling this to optimize for time to recovery of the hot ranges. Something to experiment with.

**Comment by petermattis** [comment]
> We need additional campaign logic to remove TickQuiesced completely, but until then, there's a chance it's not a win in all cases. As I said in comments in this PR, I think high latency and skewed traffic might benefit from disabling this to optimize for time to recovery of the hot ranges. Something to experiment with.

Ack.

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/22316#-:-L4M6-HcEUgTe5xx3kwj:bhm9s49)*

**Pull request merged**